File: 1200.py
class Solution(object):
    def minimumAbsDifference(self, arr):
        """
        :type arr: List[int]
        :rtype: List[List[int]]
        """
        arr.sort()
        ans = []
        min_diff = float('inf')
        
        for i in range(1, len(arr)):
            if arr[i] - arr[i-1] < min_diff:
                min_diff = arr[i] - arr[i-1]
        
        for i in range(1, len(arr)):
            if arr[i] - arr[i-1] == min_diff:
                ans.append([arr[i-1], arr[i]])
        
        return ans

File: 1200.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr);
        List<List<Integer>> ans = new ArrayList<>();
        int minDiff = Integer.MAX_VALUE;
        
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] - arr[i-1] < minDiff) {
                minDiff = arr[i] - arr[i-1];
            }
        }
        
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] - arr[i-1] == minDiff) {
                ans.add(Arrays.asList(arr[i-1], arr[i]));
            }
        }
        
        return ans;
    }
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] arr = {4, 2, 1, 3};
        List<List<Integer>> result = solution.minimumAbsDifference(arr);
        System.out.println(result);
    }
}